### DISCUSSION TITLE
Is dynamic array content available?

### DISCUSSION THREAD

**Discussion by Jacques44** [json-schema-org/community#357] [Q&A]
Hello,
please, I'd like to check the following json data and verify that "cluster" values are defined as "name" from "fields".
I'd like to use some kind of dynamic enum

Is this possible?

Thanks in advance

`{
    "fields": [
        {
            "name": "first"
        },
        {
            "name": "second"
        }
    ],
    "cluster": [
        "first"
    ]
}`

**Comment by gregsdennis** [comment]
So JSON Schema out of the box isn't able to solve this, but it may be possible with a vocabulary extension: https://json-everything.net/json-schema#schema-data-2022

To be clear, which do you want to do?

1. Ensure the items in `fields` have properties that are listed in `cluster`
2. Generate `cluster` from the values appearing in the `name` properties of the `fields` items

The first can be achieved, but not the second.

**Reply by JoshMcRose** [reply]
Hi @gregsdennis, I have a similar question [here](https://github.com/json-schema-org/community/discussions/358#discussion-4991642), but maybe it's fine to ask it here too. For my case, I want to ensure that strings listed in an array property are existing _keywords_ in my JSON file (the keywords themselves aren't predictable, but have to match a patternProperty). Is there a way to validate that using Draft-7?

**Reply by gregsdennis** [reply]
@ubertrombone you may be able to do that with my `data` vocab, but I'd need to see the specific car to be sure.

**Comment by Jacques44** [comment]
Thank you for your reply.

For example, I would like to describe a table in a database as an array of fields. Each field has a "name" property.
This is the "fields" property above.

Next, I'd like to set a clustering option that references a list of field names.
This is the "cluster" property above.

When validating my json schema, I would like to be able to check that the values of the "cluster" element are all defined as names in the "fields" document.

**Reply by gregsdennis** [reply]
Oh, so you want to ensure that all of the items in `cluster` appear as `name` properties in `fields`.

Yeah, that's not something JSON Schema can do.  You'll need something else and probably some custom code.

I suggest JSON Path to get the two lists:

- To get the `fields` values: `$.fields[*].name`
- To get the `cluster` values: `$.cluster[*]`

This will give you two lists of strings, then you just ensure that all the items in the `cluster` one appear in the `fields ` one.

**Comment by Jacques44** [comment]
ok thanks for your answers
